type-definitions/combinator/errors.d.ts

```diff
@@ -0,0 +1,6 @@
+import { Stream } from '../types';
+
+export function recoverWith<A, B>(p: (a: B) => Stream<A>, s: Stream<A>): Stream<A>;
+export function recoverWith<A, B>(p: (a: B) => Stream<A>): (s: Stream<A>) => Stream<A>;
+
+export function throwError(e: Error): Stream<any>;
```

Comment:
The flow types parameterize and constrain the error type here.  I don't see a compelling reason for it, though, other than consistency since right now, the error type doesn't really help the caller.  I could go either way.